.github/workflows/integration.yaml

```diff
@@ -7,123 +7,58 @@ on:
-        juju deploy istio-pilot --channel=1.5/stable --config default-gateway=kubeflow-gateway
-        juju relate istio-pilot:istio-pilot istio-ingressgateway:istio-pilot
-        sleep 30
-        kubectl patch role -n mlflow istio-ingressgateway-operator -p '{"apiVersion":"rbac.authorization.k8s.io/v1","kind":"Role","metadata":{"name":"istio-ingressgateway-operator"},"rules":[{"apiGroups":["*"],"resources":["*"],"verbs":["*"]}]}'
-        juju wait -wvt 300 --retry_errors 20
-        juju model-config update-status-hook-interval=5m
-
-        # https://bugs.launchpad.net/juju/+bug/1921739
-        (i=600; while ! juju wait -wvt 30 ; do ((--i)) || exit; sleep 1; done)
-
-    - name: Deploy MLflow
-      run: |
-        set -eux
-        sg microk8s -c 'juju bundle deploy --build --destructive-mode'
-        juju relate mlflow-server istio-pilot
-        juju wait -wvt 900 --retry_errors 20
-
-    - name: Test MLflow
-      run: sg microk8s -c 'tox -e selenium'
+      - uses: actions/checkout@v2
+      - run: python3 -m pip install tox
+      - run: tox -e ${{ matrix.charm }}-unit
 
-    - run: kubectl get all -A
-      if: failure()
-
-    - run: kubectl get virtualservices -A
-      if: failure()
-
-    - run: juju status
-      if: failure()
-
-    - name: Get mlflow workload logs
-      run: kubectl logs --tail 100 -nmlflow -lapp.kubernetes.io/name=mlflow-server
-      if: failure()
-
-    - name: Get mlflow operator logs
-      run: kubectl logs --tail 100 -nmlflow -loperator.juju.is/name=mlflow-server
-      if: failure()
-
-    - name: Generate inspect tarball
-      run: >
-        sg microk8s <<EOF
-          microk8s inspect | \
-          grep -Po "Report tarball is at \K.+" | \
-          xargs -I {} cp {} inspection-report-${{ strategy.job-index }}.tar.gz
-        EOF
-      if: failure()
-
-    - name: Upload inspect tarball
-      uses: actions/upload-artifact@v2
-      with:
-        name: inspection-reports
-        path: ./inspection-report-${{ strategy.job-index }}.tar.gz
-      if: failure()
-
-    - name: Upload selenium screenshots
-      uses: actions/upload-artifact@v2
-      with:
-        name: selenium-screenshots
-        path: /tmp/selenium-*.png
-      if: failure()
-
-    - name: Upload HAR logs
-      uses: actions/upload-artifact@v2
-      with:
-        name: selenium-har
-        path: /tmp/selenium-*.har
-      if: failure()
+  integration:
+    name: Integration tests (microk8s)
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        charm:
+          - mlflow-server
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup operator environment
+        uses: charmed-kubernetes/actions-operator@main
+        with:
+          provider: microk8s
+          channel: 1.21/stable
+          charmcraft-channel: latest/candidate
+      - run: sg microk8s -c 'microk8s enable metallb:10.64.140.43-10.64.140.49'
```

Comment:
Is this needed?  The actions-operator might do this for us already

Reply:
From what I can see [here](https://github.com/charmed-kubernetes/actions-operator/blob/main/src/bootstrap/index.ts#L33), it only enables dns storage and rbac

Reply:
aha!  [this](https://github.com/canonical/istio-operators/blob/a69721f08e9f627e3d29178a5f42dfcbf3405ee9/.github/workflows/integration.yaml#L43) is what I was thinking of.  No need to `run`, there's an arg instead

Reply:
I don't think this argument is available in `actions-operator` the action it is using in your example is `balchua/microk8s-actions@v0.2.2`.

Reply:
oh whoops yeah that's my fault.  Seems like something we should get into `actions-operator` then rather than `run` `microk8s` commands here